Assets/MixedRealityToolkit/Graph/README.md

```diff
@@ -0,0 +1,59 @@
+
+	Follow the [instructions](https://developer.microsoft.com/en-us/graph/docs/concepts/auth_register_app_v2) to register a **Native** application.
+
+2. Enabling .NET Framework 4.x is required.
+
+3. Declare any app capabilities required for authentication. For UWP capabilities check documentation [here](https://docs.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations).
+
+4. Identify Graph data needed for your scenario.
+
+	In the left panel of MS Graph [documentation](https://developer.microsoft.com/en-us/graph/docs/concepts/overview) identify any data and APIs needed for your scenario. For example to get the signed-in user profile the [API](https://developer.microsoft.com/en-us/graph/docs/api-reference/v1.0/api/user_get) is "/me" and "User.Read" scope is enough as stated in the documentation.
+	
+	Alternatively, use [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer) to find APIs available.
+	
+	Once you have identified all APIs and what Graph scopes (i.e. permissions) required, update the list of scopes declared in [Microsoft App Registration Portal](https://apps.dev.microsoft.com/).
+
+5. Setup Unity scene.
+
+	Add **MsalGraphConnector.cs** to any game object of your scene. Alternatively, you can add the GraphConnector.prefab to the scene, as it already includes **MsalGraphConnector.cs**.
+
+6. Setup MS Graph permissions.
+
+	Select the game object that has MsalGraphConnector.cs and update settings accordingly in the inspector.
+	* **Graph App Id** is the application ID registered in [Microsoft App Registration Portal](https://apps.dev.microsoft.com/).
+	* **Graph Access Scopes** is the array that lists all [access permissions](https://developer.microsoft.com/en-us/graph/docs/concepts/permissions_reference) required in your scenario.
+
+7. All set. 
+
+	Look up **GraphConnectorTestAsync.cs** for examples on how to use the GrahConnector.
+
+# Managed bytecode stripping with IL2CPP
+IL2CPP analyzes all assemblies and removes methods that are never directly called. If something is only accessed through reflection, it will be removed unless it is specified in link.xml files. Read more about it [here](https://docs.unity3d.com/Manual/IL2CPP-BytecodeStripping.html) 
+
+# Known issues
+Two Unity bugs prevent Microsoft.Identity.Client.dll from working in UWP:
+
+	1. IL2CPP project failing to P/Invoke into kernel32.dll!GetNativeSystemInfo. 
+
+		Issue Tracker: https://issuetracker.unity3d.com/issues/uwp-il2cpp-project-failing-to-p-slash-invoking-into-etw-logging-functions
```

Comment:
@StephenHodgson I tracked down these two bugs in Unity that are preventing MSAL from working on IL2CPP. Once these bugs are fixed, this library works.

I'd like to wrap-up this PR as soon as possible. Should we proceed with this change with that caveat (this won't work until those bugs are fixed in Unity) or do you have some other suggestion?

Reply:
I'll be sure to pull your branch and do a more in depth review this weekend.

Reply:
Hey @joaocpaiva I saw that 2018.1.6 is out with fixes for this. Going to do a dry run and see if I can get it setup.

Reply:
Tried this out but getting null references in your scene, is the auth correct for the example?

![image](https://user-images.githubusercontent.com/13334553/41812125-d25375d2-76d1-11e8-8a81-aad8a3c561e8.png)

Reply:
I'm also getting this 
![image](https://user-images.githubusercontent.com/13334553/41812328-dbe68db0-76d5-11e8-94cf-62ad141b732d.png)

Reply:
Yes, MSAL does not work in the Unity editor because the SDK requires web browser which is not available in Unity.

There are two options as workaround to test Graph in the editor:
1. Use network monitor like Fiddler, to sneak peak the Bearer token and copy/paste into GraphConnector inspector as the Test Auth Token
2. Provide custom implementation of IGraphAuthentication instead of using MsalGraphAuthentication. Obviously a real implementation is not trivial (requires OAuth, UI for users to enter credentials, etc).

Furthermore, in future releases of MSAL this scenario should be unblocked using the device code flow (which enables us to sign-in from a phone for instance), or another device. But Device code flow is not yet implemented in MSAL.Net (this is coming).

Do you have a particular concern about leaving this as is, knowing these caveats. I guess I can elaborate a bit more in the README.md?